- Provide dav1d_getauxval() wrapper for getauxvaul() and elf_aux_info()
  93f12c117a4e1c0cc2b129dcc52e84dbd9b84200
- riscv: Enable FreeBSD / OpenBSD elf_aux_info() support
  f15666b7031fa6b50f0db516d78e966acd18f5ae 

Index: src/riscv/cpu.c
--- src/riscv/cpu.c.orig
+++ src/riscv/cpu.c
@@ -32,19 +32,17 @@
 #include "src/cpu.h"
 #include "src/riscv/cpu.h"
 
-#if HAVE_GETAUXVAL
+#if HAVE_GETAUXVAL || HAVE_ELF_AUX_INFO
 #include <sys/auxv.h>
-
 #define HWCAP_RVV (1 << ('v' - 'a'))
-
 #endif
 
 int dav1d_has_compliant_rvv(void);
 
 COLD unsigned dav1d_get_cpu_flags_riscv(void) {
     unsigned flags = dav1d_get_default_cpu_flags();
-#if HAVE_GETAUXVAL
-    unsigned long hw_cap = getauxval(AT_HWCAP);
+#if HAVE_GETAUXVAL || HAVE_ELF_AUX_INFO
+    unsigned long hw_cap = dav1d_getauxval(AT_HWCAP);
     flags |= (hw_cap & HWCAP_RVV) && dav1d_has_compliant_rvv() ? DAV1D_RISCV_CPU_FLAG_V : 0;
 #endif
 
